### DISCUSSION TITLE
`white-space` div on NodeViews?

### DISCUSSION THREAD

**Discussion by WilliamIPark** [ueberdosis/tiptap#3453] [General Questions] [answered]
Hey. We're having quite a few issues stemming from this extra div added to the dom when using a NodeView, particularly around layout and animation. From my understanding the white-space rule is required, but is the approach with the additional div the optimal way to include it?

![Screenshot 2022-11-25 at 12 43 14](https://user-images.githubusercontent.com/445688/203988971-39d0a4c6-e944-42e2-baf6-6eb4458a5746.png)

**Comment by nperez0111** [comment, marked as answer]
[This is where that white-space: inherit element is created](https://github.com/ueberdosis/tiptap/blob/a42692e479200b1b15e07fc959482e6238979d4a/packages/react/src/ReactNodeViewRenderer.tsx#L82-L95), it has to do with node views' content and is the wrapper element to inject the react rendered content into. The reason that it exists is that prosemirror expects a node view to be generated synchronously, whereas react can only render asynchronously. So we inject this element, and have React render into that.

[This is where the data-node-view-content element is create](https://github.com/ueberdosis/tiptap/blob/a42692e479200b1b15e07fc959482e6238979d4a/packages/react/src/NodeViewContent.tsx#L18-L19), it is an element rendered by your react app to signal where prosemirror should inject into using that ref there. It has to exist as a handoff from React to prosemirror to not have React destroy the element that prosemirror is trying to render the rich text into.

So, all of that to say, there is no good way to get rid of that element because of how React works. You'd either have to use the JS NodeView API, or Vue because both of those have a more sane rendering approach.
